discovery/gossiper.go

```diff
@@ -3142,6 +3142,23 @@ func (d *AuthenticatedGossiper) handleChanUpdate(nMsg *networkMsg,
 		edgeToUpdate = e2
 	}
 
+	// We did a staleness check before grabbing the channelMtx mutex, but we
+	// should do another one now that the mutex is obtained in-case a
+	// duplicate or newer ChannelUpdate was processed while this thread was
+	// waiting for the lock. Even though the DB `UpdateEdge` call later on
+	// would catch this too, we need to check it now so that our rate
+	// limiting logic does not get triggered by duplicate updates.
+	if edgeToUpdate != nil && !edgeToUpdate.LastUpdate.Before(timestamp) {
```

Comment:
Or we could move `d.channelMtx.Lock` right before `if d.cfg.Graph.IsStaleEdgePolicy(`? I think `StaleEdgePolicy` also has extra logic about deciding whether an update is stale or not.

Reply:
indeed 🙈  will just move the lock :)